librarymanagement/src/main/scala/sbt/librarymanagement/UpdateOptions.scala

```diff
@@ -23,7 +23,9 @@ final class UpdateOptions private[sbt] (
     // If set to true, use cached resolution.
     val cachedResolution: Boolean,
     // Extension point for an alternative resolver converter.
-    val resolverConverter: UpdateOptions.ResolverConverter
+    val resolverConverter: UpdateOptions.ResolverConverter,
+    // Map the unique resolver to be checked for the module ID
+    val moduleResolvers: Map[ModuleID, Resolver]
```

Comment:
Yes, it should be immutable. We don't transform it internally and we don't want their contents to be exposed and modified from the outside. We just use it to represent some data that we use once inside sbt' chain resolver.